packages/pymedphys_dicom/src/pymedphys_dicom/dicom/dose.py

```diff
@@ -32,100 +30,148 @@
-            'future version this will no longer occur. To begin using this '
-            'function without the reshape pass the parameter `reshape=False` '
-            'when calling `dose_from_dataset`.'), UserWarning)
-        pixels = np.transpose(
-            ds.pixel_array, (1, 2, 0))
-    else:
-        pixels = ds.pixel_array
+        warnings.warn(
+            ('The `reshape` parameter no longer does anything. Please remove '
+             'this parameter. In a future version this parameter will no '
+             'longer be accepted.'), UserWarning)
 
-    dose = pixels * ds.DoseGridScaling
+    dose = ds.pixel_array * ds.DoseGridScaling
 
     return dose
 
 
-def axes_and_dose_from_dicom(dicom_filepath):
-    ds = pydicom.dcmread(dicom_filepath, force=True)
-    axes = xyz_axes_from_dataset(ds)
-    dose = dose_from_dataset(ds)
+def dicom_dose_interpolate(interp_coords, dose: pydicom.Dataset):
+    """Interpolates across a DICOM dose dataset.
+
+    Parameters
+    ----------
+    dose : pydicom.Dataset
+        An RT DICOM Dose object
+    interp_coords : tuple(z, y, x)
+        A tuple of coordinates in DICOM order, z axis first, then y, then x
+        where x, y, and z are DICOM axes.
+    """
+
+    interp_z = np.array(interp_coords[0], copy=False)[:, None, None]
+    interp_y = np.array(interp_coords[1], copy=False)[None, :, None]
+    interp_x = np.array(interp_coords[2], copy=False)[None, None, :]
+
+    coords, dose = zyx_and_dose_from_dataset(dose)
+    interpolation = RegularGridInterpolator(coords, dose)
+
+    try:
+        result = interpolation((interp_z, interp_y, interp_x))
+    except ValueError:
+        print(f"coords: {coords}")
+        raise
 
-    return axes, dose
+    return result
 
 
-def load_dicom_data(ds, depth_adjust):
-    dose = dose_from_dataset(ds)
-    crossplane, vertical, inplane = xyz_axes_from_dataset(ds)
+def depth_dose(depths, dose: pydicom.Dataset, plan: pydicom.Dataset):
+    depths = np.array(depths, copy=False)
 
-    depth = vertical + depth_adjust
+    surface_entry_point = get_surface_entry_point(plan)
+    depth_adjust = surface_entry_point.y
 
-    return inplane, crossplane, depth, dose
+    y = depths + depth_adjust
+    x, z = [surface_entry_point.x], [surface_entry_point.z]
 
+    coords = (z, y, x)
+
+    extracted_dose = np.squeeze(dicom_dose_interpolate(coords, dose))
+
+    return extracted_dose
+
+
+def profile(displacements, depth, direction, dose: pydicom.Dataset,
+            plan: pydicom.Dataset):
+
+    displacements = np.array(displacements, copy=False)
 
-def extract_depth_dose(ds, depth_adjust, averaging_distance=0):
-    inplane, crossplane, depth, dose = load_dicom_data(ds, depth_adjust)
+    surface_entry_point = get_surface_entry_point(plan)
+    depth_adjust = surface_entry_point.y
+    y = [depth + depth_adjust]
 
-    inplane_ref = abs(inplane) <= averaging_distance
-    crossplane_ref = abs(crossplane) <= averaging_distance
+    if direction is 'inplane':
```

Comment:
```suggestion
    if direction in ('inplane', 'inline'):
```

Reply:
Yup, fair. I don't really like this pattern I implemented here. Just couldn't seem to think of anything better... Any ideas?

Reply:
Also, of note, using `is` in and of itself was actually a bug also. See https://stackoverflow.com/a/47778970/3912576

Reply:
> Yup, fair. I don't really like this pattern I implemented here. Just couldn't seem to think of anything better... Any ideas?

Further thought, setting direction to a string and having cases based upon it isn't so bad really for this kind of function. Just not something I want to use often.

Reply:
I think it's fine :)

I guess in future it'd be nice to have an `arbitrary_profile` function that this function maybe wraps? Dunno, what do you think?

Reply:
Is the dose interpolator sort of an arbitrary profile function anyway?

Reply:
Sorry, yes it is.

It's just that the dose interpolator takes `coords` as a parameter, and I was imagining a fucntion that maybe took the two polar angles as well as `x`, `y` & `z` offsets as parameters, extracted the relevant coordinates for you and supplied them to the dose interpolator.

May not be that useful anyway.

Reply:
> I was imagining a fucntion that maybe took the two polar angles as well as `x`, `y` & `z` offsets as parameters, extracted the relevant coordinates for you and supplied them to the dose interpolator.

That would certainly be helpful! Yes, I can see that. Didn't think of that possibility.